dpos/state/arbitrators.go

```diff
@@ -1672,7 +1672,9 @@ func (a *arbitrators) snapshotVotesStates(height uint32) error {
 	nextReward.OwnerVotesInRound = make(map[common.Uint168]common.Fixed64, 0)
 	nextReward.TotalVotesInRound = 0
 	for _, ar := range a.nextArbitrators {
-		if !a.isNextCRCArbitrator(ar.GetNodePublicKey()) {
+		if !a.isNextCRCArbitrator(ar.GetNodePublicKey()) ||
+			(height >= a.chainParams.ChangeCommitteeNewCRHeight &&
+				ar.(*crcArbiter).crMember.DPOSPublicKey == nil) {
```

Comment:
`ar` can also be dposArbiter so `ca , ok := ar.*(crcArbiter) ` may be needed also cr member need to be normal or getProducer can not may be failed

Reply:
if it is noot a cr member， a.isNextCRCArbitrator(ar.GetNodePublicKey()) will be true， ar.(*crcArbiter) will be ignored.

Reply:
getProducer fixed